### PR TITLE
[Simulator] Fix TLE error when Orekit propagator is deactivated

### PR DESCRIPTION
**Pull request by X52p**
Fixes error which occurred when calling get TLE while Orekit propagator was disabled. Now returns a default TLE (all Positional values zero)

### CODE REVIEW

File: mission/simulator/opssat-spacecraft-simulator/src/main/java/opssat/simulator/threading/SimulatorNode.java [hunk 1]
```diff
@@ -291,8 +290,31 @@ public static String handleResourcePath(String path, Logger logger, ClassLoader
    */
   public TLE getTLE()
   {
-    return this.orekitCore.getTLE();
-
+    if (this.simulatorHeader.isUseOrekitPropagator()) {
+      return this.orekitCore.getTLE();
+    } else {
+      Logger.getLogger(SimulatorNode.class.getCanonicalName()).log(Level.WARNING,
```
Comment (dmarszk) [inline mission/simulator/opssat-spacecraft-simulator/src/main/java/opssat/simulator/threading/SimulatorNode.java:296]: Typo - "awaliable"

**Comment by dmarszk** [comment]
Is there nowhere to load a static TLE from instead?

**Comment by X52p** [comment]
> Is there nowhere to load a static TLE from instead?

It would be possible to replace the values with values from a real TLE, but it wouldn't be consistent with the values the simulator returns when orekit propagation is disabled. It also would be possible to load the real TLE from the internet (that has  the same problem of course)

File: mission/simulator/opssat-spacecraft-simulator/src/main/java/opssat/simulator/threading/SimulatorNode.java (hunk 1 above)
Comment (X52p) [reply mission/simulator/opssat-spacecraft-simulator/src/main/java/opssat/simulator/threading/SimulatorNode.java:296]: fixed

**Comment by dmarszk** [comment]
Let's go with the real (or realistic) TLE values. Consider the situation
where NMF is working in hybrid simulation mode with propagator disabled
(and perhaps no access to the internet), and some application depends on
the TLE

Cheers,
Dominik

W dniu czwartek, 4 czerwca 2020 Kevin Otto <notifications@github.com>
napisał(a):

**Comment by dmarszk** [comment]
On another take, I guess we can just say any experiment depending on TLE needs to enable Orekit propagator.

**Pull request merged**